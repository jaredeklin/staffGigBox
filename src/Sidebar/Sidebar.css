@import url('https://fonts.googleapis.com/css?family=Exo');

.app-title {
  font-family: 'Exo', sans-serif;
  font-size: 2.5rem;
  padding-top: 12px;
}

.log-out-container {
  display: flex;
  flex-direction: column;
}

.user-img {
  width: 60px;
  border-radius: 50%;
  margin: 30px auto 10px auto;
}

aside button {
  width: 80px;
  margin: auto;
}
