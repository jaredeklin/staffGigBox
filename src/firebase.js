import firebase from 'firebase';

const config = {
  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
  authDomain: 'staffgigbox.firebaseapp.com',
  databaseURL: 'https://staffgigbox.firebaseio.com',
  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
  storageBucket: '',
  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID
};

firebase.initializeApp(config);

export const provider = new firebase.auth.GoogleAuthProvider();
export const auth = firebase.auth();
export default firebase;
