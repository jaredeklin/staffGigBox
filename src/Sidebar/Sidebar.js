import React, { Component } from 'react';
import { auth, provider } from '../firebase.js';
import './Sidebar.css';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

export class Sidebar extends Component {
  constructor() {
    super();
    this.state = {
      user: null
    };
  }

  login = async () => {
    const result = await auth.signInWithPopup(provider);
    const { user } = await result;

    this.setState({ user });
    this.props.addUser(user);
  };

  logout = async () => {
    await auth.signOut();
    this.setState({ user: null });
    this.props.addUser(null);
  };

  componentDidMount = () => {
    auth.onAuthStateChanged(user => {
      if (user) {
        this.setState({ user });
      }

      this.props.addUser(user);
    });
  };

  render() {
    return (
      <aside>
        {this.state.user ? (
          <div className="log-out-container">
            <img
              className="user-img"
              src={this.state.user.photoURL}
              alt="profile pic"
            />
            <button onClick={this.logout}>Log Out</button>
          </div>
        ) : (
          <button onClick={this.login}>Log In</button>
        )}
        <hr />
        <ul>
          <li>
            <NavLink to="/schedule" className="nav">
              Schedule
            </NavLink>
          </li>
          <li>
            <NavLink to="/unscheduled-events" className="nav">
              Unscheduled Events
            </NavLink>
          </li>
          <li>
            <NavLink to="/availability" className="nav">
              Availability
            </NavLink>
          </li>
          <li>
            <NavLink to="/add-staff" className="nav">
              Add Staff
            </NavLink>
          </li>
          <li>
            <NavLink to="/add-events" className="nav">
              Add Events
            </NavLink>
          </li>
        </ul>
      </aside>
    );
  }
}

Sidebar.propTypes = {
  addUser: PropTypes.func
};
