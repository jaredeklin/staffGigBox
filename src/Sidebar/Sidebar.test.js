import React from 'react';
import { shallow } from 'enzyme';
import { Sidebar } from './Sidebar';
import { auth } from '../firebase.js';

describe('Sidebar', () => {
  let wrapper;
  const mockAddUser = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Sidebar addUser={mockAddUser} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should login a user', async () => {
    const result = { user: 'taco' };
    auth.signInWithPopup = jest.fn().mockReturnValue(result);

    await wrapper.instance().login();

    expect(auth.signInWithPopup).toHaveBeenCalled();
    expect(wrapper.state('user')).toEqual('taco');
    expect(mockAddUser).toHaveBeenCalledWith('taco');
  });

  it('should logout a user', async () => {
    auth.signOut = jest.fn();

    await wrapper.instance().logout();

    expect(wrapper.state('user')).toEqual(null);
    expect(mockAddUser).toHaveBeenCalledWith(null);
  });

  describe('componentDidMount', () => {
    it('should call auth.onAuthStateChanged', async () => {
      auth.onAuthStateChanged = jest.fn();

      await wrapper.instance().componentDidMount();
      expect(auth.onAuthStateChanged).toHaveBeenCalled();
    });
  });
});
