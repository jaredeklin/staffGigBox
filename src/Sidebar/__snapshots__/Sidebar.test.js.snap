// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    addUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          Log In
</button>,
        <hr />,
        <ul>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav"
                              to="/schedule"
                    >
                              Schedule
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav"
                              to="/unscheduled-events"
                    >
                              Unscheduled Events
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav"
                              to="/availability"
                    >
                              Availability
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav"
                              to="/add-staff"
                    >
                              Add Staff
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav"
                              to="/add-events"
                    >
                              Add Events
                    </NavLink>
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log In",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Log In",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav"
                            to="/schedule"
              >
                            Schedule
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav"
                            to="/unscheduled-events"
              >
                            Unscheduled Events
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav"
                            to="/availability"
              >
                            Availability
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav"
                            to="/add-staff"
              >
                            Add Staff
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav"
                            to="/add-events"
              >
                            Add Events
              </NavLink>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/schedule"
>
                Schedule
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Schedule",
                "className": "nav",
                "to": "/schedule",
              },
              "ref": null,
              "rendered": "Schedule",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/unscheduled-events"
>
                Unscheduled Events
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Unscheduled Events",
                "className": "nav",
                "to": "/unscheduled-events",
              },
              "ref": null,
              "rendered": "Unscheduled Events",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/availability"
>
                Availability
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Availability",
                "className": "nav",
                "to": "/availability",
              },
              "ref": null,
              "rendered": "Availability",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/add-staff"
>
                Add Staff
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Add Staff",
                "className": "nav",
                "to": "/add-staff",
              },
              "ref": null,
              "rendered": "Add Staff",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/add-events"
>
                Add Events
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Add Events",
                "className": "nav",
                "to": "/add-events",
              },
              "ref": null,
              "rendered": "Add Events",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            Log In
</button>,
          <hr />,
          <ul>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav"
                                    to="/schedule"
                        >
                                    Schedule
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav"
                                    to="/unscheduled-events"
                        >
                                    Unscheduled Events
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav"
                                    to="/availability"
                        >
                                    Availability
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav"
                                    to="/add-staff"
                        >
                                    Add Staff
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="nav"
                                    to="/add-events"
                        >
                                    Add Events
                        </NavLink>
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log In",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Log In",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav"
                                to="/schedule"
                >
                                Schedule
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav"
                                to="/unscheduled-events"
                >
                                Unscheduled Events
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav"
                                to="/availability"
                >
                                Availability
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav"
                                to="/add-staff"
                >
                                Add Staff
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav"
                                to="/add-events"
                >
                                Add Events
                </NavLink>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/schedule"
>
                  Schedule
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Schedule",
                  "className": "nav",
                  "to": "/schedule",
                },
                "ref": null,
                "rendered": "Schedule",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/unscheduled-events"
>
                  Unscheduled Events
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Unscheduled Events",
                  "className": "nav",
                  "to": "/unscheduled-events",
                },
                "ref": null,
                "rendered": "Unscheduled Events",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/availability"
>
                  Availability
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Availability",
                  "className": "nav",
                  "to": "/availability",
                },
                "ref": null,
                "rendered": "Availability",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/add-staff"
>
                  Add Staff
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Add Staff",
                  "className": "nav",
                  "to": "/add-staff",
                },
                "ref": null,
                "rendered": "Add Staff",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/add-events"
>
                  Add Events
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Add Events",
                  "className": "nav",
                  "to": "/add-events",
                },
                "ref": null,
                "rendered": "Add Events",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
